### PR TITLE
New gpbackup options

### PR DESCRIPTION
**Pull request by dyozie**
All changes will also go into master branch.

HTML pages for these changes are available at:

http://docs-gpdb-review-staging.cfapps.io/530/admin_guide/managing/backup-gpbackup.html
http://docs-gpdb-review-staging.cfapps.io/530/utility_guide/admin_utilities/gpbackup.html
http://docs-gpdb-review-staging.cfapps.io/530/utility_guide/admin_utilities/gprestore.html

### CODE REVIEW

File: gpdb-doc/dita/utility_guide/admin_utilities/gpbackup.xml [hunk 1]
```diff
@@ -74,6 +76,17 @@
             the backup directory.</pd>
         </plentry>
       </parml>
+      <parml>
+        <plentry>
+          <pt><b>-compression-level</b>
+            <varname>level</varname></pt>
+          <pd>Optional. Specifies the gzip compression level (from 1 to 9) used to compress data
+            files. The default is 1.  Note that <codeph>gpbackup</codeph> uses compression by
+            default. If you specify both <codeph>-compression-level</codeph> and
```
Comment (jmcatamney) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpbackup.xml]: This is incorrect; the `compression-level` and `no-compression` flags cannot be specified together.

File: gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml [hunk 2]
```diff
@@ -292,6 +296,35 @@ public.sales_1_prt_dec17</codeblock>
...
+          options work in the same way as their <codeph>gpbackup</codeph> counterparts, but have the
+          following restrictions:<ul id="ul_hts_clc_ccb">
+            <li>The schemas or tables that you attempt to restore must not already exist in the
+              database.</li>
+            <li>If you attempt to restore a schema or table that does not exist in the backup set,
+              the <codeph>gprestore</codeph> does not execute.</li>
+            <li>Backup sets that use a single data file for all tables on a segment
+                (<codeph>gpbackup -single-data-file</codeph> option) cannot be restored using the
+                <codeph>-include-schema</codeph> option.</li>
+            <li>You cannot use the <codeph>-include-schema</codeph> and  <codeph>-redirect</codeph>
```
Comment (chrishajas) [inline gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml]: We do allow and support the -redirect and -include-schema flags together

File: gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml [hunk 3]
```diff
@@ -292,6 +296,35 @@ public.sales_1_prt_dec17</codeblock>
...
+              options in the same <codeph>gprestore</codeph> command.</li>
+            <li>If you use the <codeph>-include-schema</codeph> option, <codeph>gprestore</codeph>
+              cannot restore objects that have dependencies on multiple schemas.</li>
+            <li>If you use the <codeph>-include-table-file</codeph> option,
+                <codeph>gprestore</codeph> does not restore indexes, create roles, or set the owner
+              of the tables. <codeph>gprestore</codeph> also does not restore global metadata, so
```
Comment (chrishajas) [inline gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml]: We do allow and support -include-table-file and -globals together.

File: gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml [hunk 4]
```diff
@@ -438,6 +473,9 @@ public.sales_1_prt_dec17</codeblock>
                 <filepath>&lt;oid></filepath> to locate the data for a specific table in a
               schema.</li>
           </ul></p>
+        <p>You can optionally specify the gzip compression level (from 1-9) using the
```
Comment (chrishajas) [inline gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml]: If no specific compression level is specified, we use compression level 1.

**Comment by dyozie** [comment]
Thanks for the feedback - all comments so far have been addressed in new commits.  Merging this PR for now, but feel free to add more comments if you notice any other issues.

**Pull request merged**